statusHandler/persister/persistentHandler.go

```diff
@@ -0,0 +1,200 @@
+package persister
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/typeConverters"
+	"github.com/ElrondNetwork/elrond-go/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+var log = logger.GetOrCreate("statusHandler/persister")
+
+// PersistentStatusHandler is a status handler that will save metrics in storage
+type PersistentStatusHandler struct {
+	store                    storage.Storer
+	persistentMetrics        *sync.Map
+	marshalizer              marshal.Marshalizer
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
+	startSaveInStorage       bool
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+	uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter,
+) (*PersistentStatusHandler, error) {
+	if check.IfNil(marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(uint64ByteSliceConverter) {
+		return nil, process.ErrNilUint64Converter
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
```

Comment:
Why do you need this NilStorer?

Reply:
NilStorer is needded because i cannot use a storer component when PersistentStatusHandler is created because in main.go first time is created coreComponents that contains status handlers and after that are created dataComponents that contains storer. And i can use a storer just after dataComponents are created.

Reply:
And you update psh.store after the dataComponents are created?

Reply:
Have you updated psh.store after the dataComponents are created with the real storer?